microsoft/testsuites/core/cpu.py

```diff
@@ -31,8 +31,9 @@ class CPU(testsuite.TestSuite):
     def _set_cpu_state(self, cpu_id: str, state: str, node: Node) -> bool:
         file_path = f"/sys/devices/system/cpu/cpu{cpu_id}/online"
         file_exists = node.shell.exists(PurePosixPath(file_path))
-        if file_exists:
-            node.tools[Echo].write_to_file(state, file_path, shell=True, sudo=True)
+        if not file_exists:
```

Comment:
what's the reason the file is not here? Should the test case be skipped, or just skip some of cpus?

Reply:
The file will not be present if the CPU is reserved for critical OS features. For example, in smaller VM's CPU0 could not be offline. The same thing happens with other CPU's on larger VM's such as `Standard_E96as_v4`

Reply:
It happens on some of CPUs or all of them? Can we skip it, if it applies to all of CPUs?

Reply:
I didn't come across any VM which has all it's CPU's reserved. And I don't think we should skip this test. The intent of the test is to check that all the CPU's assigned to lsvmbus could not be offline. In this scenario, the CPU's are still assigned to lsvmbus even if they are reserved.

Reply:
The test should be skipped if the conditions could be not be verified or favorable to run test. But in this scenario, we are able to verify that all the cpu's mapped to lsvmbus could not be offline.

Reply:
My understanding is the goal is to check the cpu cannot be set to offline by `echo 0 > online`, and check its value. If there is no chance to set it on every cpu. It shouldn't be covered. Maybe some distro or cpu doesn't support this online/offline switch, but it may support another way.